pupil_src/shared_modules/surface_tracker/surface_tracker_offline.py

```diff
@@ -0,0 +1,785 @@
+                all_fixation_events,
+                self.camera_model,
+            )
+
+    def gl_display(self):
+        if self.timeline:
+            self.timeline.refresh()
+        super().gl_display()
+
+    def _gl_display_cache_bars(self, width, height, scale):
+        ts = self.g_pool.timestamps
+        with gl_utils.Coord_System(ts[0], ts[-1], height, 0):
+            # Lines for areas that have been cached
+            cached_ranges = []
+            for r in self.marker_cache.visited_ranges:
+                cached_ranges += ((ts[r[0]], 0), (ts[r[1]], 0))
+
+            gl.glTranslatef(0, scale * self.timeline_line_height / 2, 0)
+            color = pyglui_utils.RGBA(0.8, 0.2, 0.2, 0.8)
+            pyglui_utils.draw_polyline(
+                cached_ranges, color=color, line_type=gl.GL_LINES, thickness=scale * 4
+            )
+            cached_ranges = []
+            for r in self.marker_cache.positive_ranges:
+                cached_ranges += ((ts[r[0]], 0), (ts[r[1]], 0))
+
+            color = pyglui_utils.RGBA(0, 0.7, 0.3, 0.8)
+            pyglui_utils.draw_polyline(
+                cached_ranges, color=color, line_type=gl.GL_LINES, thickness=scale * 4
+            )
+
+            # Lines where surfaces have been found in video
+            cached_surfaces = []
+            for surface in self.surfaces:
+                found_at = []
+                if surface.location_cache is not None:
+                    for r in surface.location_cache.positive_ranges:  # [[0,1],[3,4]]
+                        found_at += ((ts[r[0]], 0), (ts[r[1]], 0))
+                    cached_surfaces.append(found_at)
+
+            color = pyglui_utils.RGBA(0, 0.7, 0.3, 0.8)
+
+            for surface in cached_surfaces:
+                gl.glTranslatef(0, scale * self.timeline_line_height, 0)
+                pyglui_utils.draw_polyline(
+                    surface, color=color, line_type=gl.GL_LINES, thickness=scale * 2
+                )
+
+    def _draw_labels(self, width, height, scale):
+        self.glfont.set_size(self.timeline_line_height * 0.8 * scale)
+        self.glfont.draw_text(width, 0, "Marker Cache")
+        for idx, s in enumerate(self.surfaces):
+            gl.glTranslatef(0, self.timeline_line_height * scale, 0)
+            self.glfont.draw_text(width, 0, s.name)
+
+    def add_surface(self, _=None, init_dict=None):
+        super().add_surface(init_dict=init_dict)
+        # Plugin initialization loads surface definitions before UI is initialized. Changing timeline height will fail in this case.
+        if self.markers or init_dict is not None:
+            try:
+                self.timeline.content_height += self.timeline_line_height
+                self._fill_gaze_on_surf_buffer()
+            except AttributeError:
+                pass
+            self.surfaces[-1].on_surface_changed = self.on_surface_change
+
+    def remove_surface(self, _):
+        super().remove_surface(_)
```

Comment:
Imo, the convention is that if one uses `_` as an argument, one is not interested its value. However, in this case you are interested because you need to pass it to the super call. I would suggest to rename it to something meaningful. Any opinion on this @papr ?